modules/structs/repo_compare.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2018 The Gogs Authors. All rights reserved.
```

Comment:
did you copy this files from gogs?

Reply:
already remove the line.

Reply:
@Madongming you should not remove the line if the code comes from Gogs, as attribution is due to each author. You should instead add a line for Gitea authors, as you can find in other source files. For example:

https://github.com/go-gitea/gitea/blob/e008b68c9c312681d4098f6f3b8f208479f4f43a/models/action.go#L1-L6

Reply:
this is why I asked **if** the code is from gogs or if you programmed it yourselve ...
@Madongming :point_up:

Reply:
I copied a lot of code from the file gitea/routers/repo/.go, So I used this file header for Copyright